securebanking-openbanking-uk-gateway-conversion-filter/src/main/java/com/forgerock/securebanking/uk/gateway/conversion/filter/IntentConverterFilter.java

```diff
@@ -0,0 +1,287 @@
+import static org.forgerock.json.JsonValueFunctions.enumConstant;
+import static org.forgerock.openig.el.Bindings.bindings;
+import static org.forgerock.openig.util.AsyncFunctions.asyncResultHandler;
+import static org.forgerock.util.promise.NeverThrowsException.neverThrown;
+import static org.forgerock.util.promise.Promises.newResultPromise;
+
+/**
+ * Filter to convert IDM json intent objects to OB data model objects.
+ *
+ * This filter must have received {@code intentType} as required to identify the intent type {@link IntentType#toString()} to instance the converter<br/>
+ * Failures from the `Converter instance` is either a {@link RuntimeException} that will catch to build a {@link ResponseException}.
+ *
+ * Configuration options:
+ *
+ * <pre>
+ * {@code {
+ *      "name": "IntentConverterFilter-name"
+ *      "type": "IntentConverterFilter",
+ *      "config": {
+ *         "messageType"  MessageType              [REQUIRED - The type of message for which to convert the entity, Must be either "REQUEST" or "RESPONSE"]
+ *         "entity"       Expression<String>       [OPTIONAL - A jsonPayload content to be converted in string format. Default : not replaced.]
+ *         "resultTo"     List<MessageType>        [OPTIONAL - Indicates where will set the conversion result. Must be either "REQUEST" or "RESPONSE". Default REQUEST.]
+ *      }
+ *  }
+ *  }
+ * </pre>
+ * <p>Example</p>
+ * <pre>
+ * {@code {
+ *      "name": "IntentConverterFilter-name"
+ *      "type": "IntentConverterFilter",
+ *      "config": {
+ *         "messageType: "request",
+ *         "entity": "#{request.entity.string}",
+ *         "resultTo": ["REQUEST", "RESPONSE],
+ *      }
+ *  }
+ *  }
+ * </pre>
+ */
+public class IntentConverterFilter implements Filter {
+
+    private static final Logger logger = LoggerFactory.getLogger(IntentConverterFilter.class);
+
+    private final MessageType messageType;
+    private final List<MessageType> resultTo;
+    private final Expression<String> entity;
+    private static final ObjectMapper MAPPER = GenericConverterMapper.getMapper();
+
+    /**
+     * Constructor
+     * @param messageType
+     * @param entity
+     * @param resultTo
+     */
+    public IntentConverterFilter(MessageType messageType, final Expression<String> entity, final List<MessageType> resultTo) {
+        this.messageType = messageType;
+        this.entity = entity;
+        this.resultTo = resultTo;
+    }
+
+    // Constructors for test support
+    @VisibleForTesting
+    IntentConverterFilter(MessageType messageType) {
+        this(messageType, null, List.of(MessageType.REQUEST));
+    }
+
+    @VisibleForTesting
+    IntentConverterFilter(MessageType messageType, final Expression<String> entity) {
+        this(messageType, entity, List.of(MessageType.REQUEST));
+    }
+
+    @VisibleForTesting
+    IntentConverterFilter(MessageType messageType, final List<MessageType> resultTo) {
+        this(messageType, null, resultTo);
+    }
```

Comment:
these constructors don't add any functionality which can't be achieved through the public constructor, therefore I think it would be cleaner to create factory methods under src/test for these rather than telescoping the constructor solely for testing purposes.

Reply:
Constructors for test deleted, created a factory test filter to get the instance of filter.